metricbeat/module/redis/redis.go

```diff
@@ -76,6 +76,97 @@ func FetchSlowLogLength(c rd.Conn) (int64, error) {
 	return count, nil
 }
 
+// FetchKeyInfo collects info about a key
+func FetchKeyInfo(c rd.Conn, key string) (map[string]interface{}, error) {
+	keyType, err := rd.String(c.Do("TYPE", key))
+	if err != nil {
+		return nil, err
+	}
+	if keyType == "none" {
+		// Ignore it, it has been removed
+		return nil, nil
+	}
+
+	keyTTL, err := rd.Int64(c.Do("TTL", key))
+	if err != nil {
+		return nil, err
+	}
+
+	info := map[string]interface{}{
```

Comment:
Why not use common.MapStr here?

Reply:
This file was not using `MapStr` or even `common` at the moment, I didn't want to add it now, I don't see it will bring anything here.

Reply:
Using constants for type names now.